hoomd/md/pair/pair.py

```diff
@@ -452,6 +452,95 @@ def __init__(self, nlist, default_r_cut=None, default_r_on=0., mode='none'):
+    type pair potential should be applied between every non-excluded
+    particle pair in the simulation.
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        V_{\mathrm{ExpandedLJ}}(r)  = & 4 \varepsilon \left[ \left(
+                                \frac{\sigma}{r - \Delta} \right)^{12} -
+                                \left( \frac{\sigma}{r - \Delta}
+                                \right)^{6} \right]; & r < (r_{\mathrm{cut}}) \\
+                             = & 0; & r \ge (r_{\mathrm{cut}}) \\
+        \end{eqnarray*}
+
+    See `Pair` for details on how forces are calculated and the
+    available energy shifting and smoothing modes.
+
+    Attention:
+        Due to the way that `ExpandedLJ` modifies the cutoff criteria, a smoothing mode
+        of *xplor* is not supported.
+
+    Set the ``max_diameter`` property of the neighbor list object to the largest
+    particle diameter in the system (where **diameter** is a per-particle
+    property of the same name in `hoomd.State`). TODO: check if needed this warning
+
+    Warning:
+        Failure to set ``max_diameter`` will result in missing pair
+        interactions.
+
+    .. py:attribute:: params
+
+        The potential parameters. The dictionary has the following keys:
+
+        * ``epsilon`` (`float`, **required**) - energy parameter
+          :math:`\varepsilon` :math:`[\mathrm{energy}]`
+        * ``sigma`` (`float`, **required**) - particle size :math:`\sigma`
+          :math:`[\mathrm{length}]`
+        * ``delta`` (`float`, **required**) - horizontal shift
+          :math:`\Delta` :math:`[\mathrm{length}]`
+
+        Type: `TypeParameter` [`tuple` [``particle_type``, ``particle_type``],
+        `dict`]
+
+    Example::
+
+        nl = nlist.Cell()
+        nl.max_diameter = 2.0
+        slj = pair.SLJ(default_r_cut=3.0, nlist=nl)
+        slj.params[('A', 'B')] = dict(epsilon=2.0, r_cut=3.0)
+        slj.r_cut[('B', 'B')] = 2**(1.0/6.0)
+
+    """
+    _cpp_class_name = 'PotentialPairExpandedLJ'
+
+    def __init__(self, nlist, default_r_cut=None, default_r_on=0., mode='none'):
+        if mode == 'xplor':
+            raise ValueError(
+                "xplor is not a valid mode for ExpandedLJ potential")
+
+        super().__init__(nlist, default_r_cut, default_r_on, mode)
+        params = TypeParameter(
+            'params', 'particle_types',
+            TypeParameterDict(epsilon=float,
+                              sigma=float,
+                              delta=float,
+                              len_keys=2))
+        self._add_typeparam(params)
+
+        # mode not allowed to be xplor, so re-do param dict entry without that
+        # option
+        param_dict = ParameterDict(mode=OnlyFrom(['none', 'shift']))
+        self._param_dict.update(param_dict)
+        self.mode = mode
+
+        # this potential needs diameter shifting on
+        self._nlist.diameter_shift = True
```

Comment:
Do we still need `self._nlist.diameter_shift` set to True?

Reply:
Yes. We want to find neighbors properly subject to the new, more general form of the delta parameter.

Reply:
Except that the more general form requires the user to choose `r_cut` appropriately. Remove the `diameter_shift = True` setting.